### PR TITLE
397: Null Safety all the way! (almost)

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

File: frontend/lib/about/about_page.dart [hunk 1]
```diff
@@ -9,17 +9,24 @@ import 'dev_settings_view.dart';
...
-  const AboutPage({Key key}) : super(key: key);
+  const AboutPage({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     var config = Configuration.of(context);
+
+    if (config == null) {
+      throw Exception("Could not find configuration in component tree.");
+    }
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/about/about_page.dart:20]: done

File: frontend/lib/about/about_page.dart [hunk 2]
```diff
@@ -36,14 +43,16 @@ class AboutPage extends StatelessWidget {
...
-                ),
-                Center(
-                  child: Text(snapshot.data.version,
-                      style: Theme.of(context).textTheme.bodyText2),
-                ),
+                if (packageInfo != null)
+                  Center(
+                    child: Text(packageInfo.appName,
+                        style: Theme.of(context).textTheme.headline5),
+                  ),
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/about/about_page.dart:50]: done

File: frontend/lib/identification/identification_qr_content_parser.dart [hunk 3]
```diff
@@ -69,9 +71,10 @@ class IdentificationQrContentParser {
     if (!cardActivateModel.hasTotpSecret()) {
       throw QrCodeFieldMissingException("totpSecret");
     }
-    String base32TotpSecret;
+    String? base32TotpSecret;
     try {
-      base32TotpSecret = base32.encode(cardActivateModel.totpSecret);
+      base32TotpSecret = base32.encode(Uint8List.fromList(cardActivateModel
+          .totpSecret)); // FIXME: Changed code here in unsafe migration
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/identification/identification_qr_content_parser.dart:77]: done

File: frontend/lib/map/map/map.dart [hunk 4]
```diff
@@ -58,96 +57,119 @@ class _MapState extends State<Map> implements MapController {
...
-                },
-              );
-            },
-          ),
-        ),
-      ]);
+
+    if (_isAnimating) {
+      _mapboxView = null;
     }
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/map/map/map.dart:73]: thats like it was before, not sure what it is doing

File: frontend/lib/search/filter_bar_button.dart [hunk 5]
```diff
@@ -21,8 +22,8 @@ class FilterBarButton extends StatefulWidget {
 class _FilterBarButtonState extends State<FilterBarButton>
     with SingleTickerProviderStateMixin {
   bool _selected = false;
-  AnimationController _animationController;
-  Animation _colorTween;
+  AnimationController? _animationController;
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/search/filter_bar_button.dart:25]: I would only use late if it is initalized in the constructor.

File: frontend/lib/application/application_form.dart [hunk 6]
```diff
@@ -26,7 +26,7 @@ class _ApplicationFormState extends State<ApplicationForm> {
   bool _sendingSuccessful = false;
 
   int _currentStep = 0;
-  GraphQLClient _client;
+  GraphQLClient? _client;
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/application/application_form.dart:29]: I dont think so. I would use late only when initializing in constructors.

File: frontend/lib/about/texts.dart [hunk 7]
```diff
@@ -4,17 +4,20 @@ import 'package:flutter/material.dart';
...
-            if (e.title != null)
-              Text(e.title, style: theme.textTheme.headline6),
-            Text(e.content, style: theme.textTheme.bodyText1)
-          ])
+      .map((e) {
+        var title = e.title;
+        var content = e.content;
+        return [
+          if (title != null) Text(title, style: theme.textTheme.headline6),
+          if (content != null) Text(content, style: theme.textTheme.bodyText1)
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/about/texts.dart:18]: yes, dart is very shitty here imo

File: frontend/lib/application/application_form.dart [hunk 8]
```diff
@@ -107,18 +107,29 @@ class _ApplicationFormState extends State<ApplicationForm> {
     // can go forward by tapping only if the forms in between are valid
     if (step > _currentStep) {
       for (var i = _currentStep; i < step; i++) {
-        if (!_formKeys[i].currentState.validate()) return;
+        var currentState = _formKeys[i].currentState;
+        
+        if (currentState == null) {
+          continue;
+        }
```
[the remark this answers is not in the document]
Comment (maxammann) [reply frontend/lib/application/application_form.dart:114]: I dont know :D

**Comment by maxammann** [comment]
@Schedulaar I resolved all items from the `application` package, because it will be removed soon.

File: frontend/lib/search/filter_bar_button.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (Schedulaar) [reply frontend/lib/search/filter_bar_button.dart:25]: if it was initialized in the constructor, we wouldn't need late I guess?

File: frontend/lib/map/map/map.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (Schedulaar) [reply frontend/lib/map/map/map.dart:67]: nope

**Pull request merged**